docs/resources/source_table_kafka.md

```diff
@@ -0,0 +1,325 @@
+subcategory: ""
+description: |-
+  A Kafka source describes a Kafka cluster you want Materialize to read data from.
+---
+
+# materialize_source_table_kafka (Resource)
+
+A Kafka source describes a Kafka cluster you want Materialize to read data from.
+
+
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) The identifier for the source.
+- `source` (Block List, Min: 1, Max: 1) The source this table is created from. (see [below for nested schema](#nestedblock--source))
+- `upstream_name` (String) The name of the table in the upstream database.
+
+### Optional
+
+- `comment` (String) **Public Preview** Comment on an object in the database.
+- `database_name` (String) The identifier for the source database in Materialize. Defaults to `MZ_DATABASE` environment variable if set or `materialize` if environment variable is not set.
+- `envelope` (Block List, Max: 1) How Materialize should interpret records (e.g. append-only, upsert).. (see [below for nested schema](#nestedblock--envelope))
+- `expose_progress` (Block List, Max: 1) The name of the progress collection for the source. If this is not specified, the collection will be named `<src_name>_progress`. (see [below for nested schema](#nestedblock--expose_progress))
+- `format` (Block List, Max: 1) How to decode raw bytes from different formats into data structures Materialize can understand at runtime. (see [below for nested schema](#nestedblock--format))
+- `include_headers` (Boolean) Include message headers.
+- `include_headers_alias` (String) Provide an alias for the headers column.
+- `include_key` (Boolean) Include a column containing the Kafka message key.
+- `include_key_alias` (String) Provide an alias for the key column.
+- `include_offset` (Boolean) Include an offset column containing the Kafka message offset.
+- `include_offset_alias` (String) Provide an alias for the offset column.
+- `include_partition` (Boolean) Include a partition column containing the Kafka message partition
+- `include_partition_alias` (String) Provide an alias for the partition column.
+- `include_timestamp` (Boolean) Include a timestamp column containing the Kafka message timestamp.
+- `include_timestamp_alias` (String) Provide an alias for the timestamp column.
+- `key_format` (Block List, Max: 1) Set the key format explicitly. (see [below for nested schema](#nestedblock--key_format))
+- `ownership_role` (String) The owernship role of the object.
+- `region` (String) The region to use for the resource connection. If not set, the default region is used.
+- `schema_name` (String) The identifier for the source schema in Materialize. Defaults to `public`.
+- `start_offset` (List of Number) Read partitions from the specified offset.
+- `start_timestamp` (Number) Use the specified value to set `START OFFSET` based on the Kafka timestamp.
+- `upstream_schema_name` (String) The schema of the table in the upstream database.
```

Comment:
what does this refer to for kafka sources? we might just want to omit it since the upstream reference should just be the kafka topic name

Reply:
Good catch, this was an overlook on my end in the schema for the Kafka source table resource.